Sources/CacheAdvance/CacheHeaderHandle.swift

```diff
@@ -83,60 +74,55 @@ final class CacheHeaderHandle {
         try handle.write(data: currentHeader.data(for: .offsetInFileAtEndOfNewestMessage))
     }
 
-    /// Checks if the static header metadata of this object matches another file header. Static metadata are the properties of the header
-    /// that need to remain the same for the entire lifetime of a cache file.
-    ///
-    /// - Parameter fileHeader: A representation of the header data in an existing cache file.
+    /// Checks if the expected header version matches the persisted header version.
     ///
-    /// - Returns: `true` if this object's static metadata matches that of `fileHeader`; otherwise `false`.
-    func canOpenFile(with fileHeader: FileHeader) -> Bool {
-        guard fileHeader.version == version else {
-            // Our current file header version is 1.
-            // That means there is no prior header version we could attempt to read.
-            // We have no idea how to read this file.
-            return false
-        }
-
-        guard
-            fileHeader.maximumBytes == maximumBytes,
-            fileHeader.overwritesOldMessages == overwritesOldMessages
-            else
-        {
-            // The header's values are not consistent with our expectations.
-            return false
-        }
+    /// - Returns: `true` if this object's header version matches that of `fileHeader`; otherwise `false`.
+    func canOpenFile() throws -> Bool {
+        canOpenFile(with: try memoizedMetadata())
+    }
 
-        return true
+    /// Checks if the all the header metadata provided at initialization matches the persisted header.
+    ///
+    /// - Returns: `true` if this object's static metadata matches that of the persisted `fileHeader`; otherwise `false`.
+    func canWriteToFile() throws -> Bool {
+        canWriteToFile(with: try memoizedMetadata())
     }
 
     /// Reads the header data from the file. Writes header information to disk if no header exists.
     /// If the header data persisted to the file is not consistent with expectations, the file will be deleted.
     func synchronizeHeaderData() throws {
         let headerData = try readHeaderData()
 
-        guard let fileHeader = FileHeader(from: headerData) else {
-            // We can't read the header data. Let's start over.
-            try resetFile()
+        guard !headerData.isEmpty else {
+            // The file is empty. Write a header to disk.
+            try writeHeaderData()
+            persistedMetadata = Metadata(fileHeader: currentHeader)
             return
         }
 
-        guard canOpenFile(with: fileHeader) else {
-            // The header is invalid. Nuke it.
-            try resetFile()
-            return
+        guard let fileHeader = FileHeader(from: headerData) else {
+            // We can't read the header data.
+            throw CacheAdvanceError.fileCorrupted
```

Comment:
We'd recover by deleting the cache. I'm trying to move towards a world where we no longer delete data on disk, and instead report what went wrong and allow our consumers to handle as they please.

Now that I'm trying to integrate CacheAdvance into our app, I'm realizing that auto-delete isn't really acceptable, even if we're super clear when it would happen.

Reply:
> Now that I'm trying to integrate CacheAdvance into our app, I'm realizing that auto-delete isn't really acceptable, even if we're super clear when it would happen.

Yeah I tend to agree.